File: sic-api/Entities/Su/SuBusinessRoleProgram.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace sic_api.Entities.Su;

[Table("su_business_role_program")]
public class SuBusinessRoleProgram : BaseEntity
{
    [Required]
    [ForeignKey(nameof(BusinessRole))]
    [Column("business_role_id")]
    public Guid BusinessRoleId { get; set; }
    public SuBusinessRole BusinessRole { get; set; } = default!;

    [Required]
    [ForeignKey(nameof(Program))]
    [Column("program_id")]
    public Guid ProgramId { get; set; }
    public SuProgram Program { get; set; } = default!;

    [Column("is_active")]
    public bool IsActive { get; set; } = false;

    [Column("role_back")]
    public bool RoleBack { get; set; } = false;

    [Column("role_search")]
    public bool RoleSearch { get; set; } = false;

    [Column("role_add")]
    public bool RoleAdd { get; set; } = false;

    [Column("role_save")]
    public bool RoleSave { get; set; } = false;

    [Column("role_delete")]
    public bool RoleDelete { get; set; } = false;

    [Column("role_print")]
    public bool RolePrint { get; set; } = false;

    
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/su/SuBusinessRoleProgram.java
package com.softinter.sicapi.entity.su;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Table(name = "su_business_role_program",
       indexes = {
           @Index(name = "idx_business_role_program", columnList = "business_role_id, program_id", unique = true)
       })
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class SuBusinessRoleProgram extends BaseEntity {

    @Column(name = "business_role_id", insertable = false, updatable = false)
    private UUID businessRoleId;

    @Column(name = "program_id", insertable = false, updatable = false)
    private UUID programId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "business_role_id", nullable = false)
    private SuBusinessRole businessRole;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "program_id", nullable = false)
    private SuProgram program;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;

    @Column(name = "is_add")
    private boolean isAdd;

    @Column(name = "is_back")
    private boolean isBack;

    @Column(name = "is_print")
    private boolean isPrint;

    @Column(name = "is_remove")
    private boolean isRemove;

    @Column(name = "is_save")
    private boolean isSave;

    @Column(name = "is_search")
    private boolean isSearch;
}